ls1intum/Artemis
# Migrate Existing Feedback for Text Exercises to Text Blocks

### Checklist

### Motivation and Context
With the Introduction of Automatic Feedback Suggestions for Text Exercises, we introduced the `TextBlock` Entity. This provides a way of referencing a pice of text by its index. This also circumvents certain problems occurring when referencing identical text twice in a single assessment. Before, text was referenced by simply copying the reference text.

### Description
This PR provides a Migration for converting existing `Feedback` to the new `TextBlock` based representation.

### Steps for Testing


1. Before starting Artemis, find existing text feedback in the database and see the `reference` column in the `feedback` table.
2. Start Artemis.
3. Check again in database to find sha1 hashes instead of text in `feedback` table.
4. Check the `text_block` table to find a those same hashes with start/end index and submission reference.

---

**⚠️ This should not be merged as long as new Feedback is saved in the old format. PR incomming.**

~So far, this PR does not contain JavaDoc comments. However, these changes will be reverted as soon as it was run on the production instance, so we can save those efforts towards more useful documentation efforts like #717.~
